request/executor.go

```diff
@@ -7,44 +7,85 @@ import (
 )
 
 // ExecuteRequest executes an HTTP request based on the specified options.
-func ExecuteRequest(request Request, profileNames []string, variables map[string]string) (*ExecutedRequestResponse, error) {
-	httpRequest, configuredRequest, httpRequestErr := BuildRequest(request, profileNames, variables)
-	if httpRequestErr != nil {
-		return nil, httpRequestErr
-	}
+func ExecuteRequest(request Request, profileNames []string, variables map[string]string) ([]ExecutedRequestResponse, error) {
+	toExecute := make([]Request, 1)
+	toExecute[0] = request
 
-	client := &http.Client{}
-	httpResponse, httpResponseErr := client.Do(httpRequest)
-	if httpResponseErr != nil {
-		return nil, httpResponseErr
+	client := &http.Client{
```

Comment:
I'm not sure [the default client has a request timeout](https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779)? Maybe add this here?

Looks like it does have a default policy around consecutive requests so no infinite redirects should happen
![image](https://user-images.githubusercontent.com/8136030/34802650-b8b2d8cc-f63c-11e7-8d4b-1ec5ea0cf32f.png)

Reply:
I thought about adding logic to count redirects and break if reaches a max. Didn't add it because I was lazy. I'll add it.

Reply:
I don't know if you missed it, but I think you should add a request timeout like
```golang
var netClient = &http.Client{
  Timeout: time.Second * 10,
}
```
because I think the default value is `0` which is interpreted as "no timeout" so theoretically, the process could hang for a long time?

Reply:
I missed it. But now that I'm thinking about it, it's nice to hang while you're debugging some code after executing a request. So for now, I'll leave it without a timeout.